packages/back-end/src/controllers/datasources.ts

```diff
@@ -428,11 +428,11 @@ export async function putDataSource(
   req: AuthRequest<
     {
       name: string;
-      description?: string;
+      description: string;
```

Comment:
There is only one place in the app where we're hitting this endpoint, and we're passing in the full "Datasource" object, not just the updates, so description will always be provided, it may just be an empty string. Same with projects

Reply:
Is this true?  It looks like it's also called in `[did].tsx` where only `settings` and nothing else is passed in.

Reply:
I don't know how I missed that. I've updated this PR in light of this. I mostly reset everything, and when it comes to the description, I'm checking to see if the `description` property exists in the `req.body`, and if so, then setting using the value to update the description.